src/hooks/useLoggedInUser.js

```diff
@@ -3,22 +3,20 @@
 import { useState, useEffect } from 'react';
 
 const useLoggedInUser = defaultName => {
-    const [user, setUser] = useState({ name: extractName(defaultName) });
+    const [userNameFromCookie, setUserNameFromCookie] = useState(extractName());
 
     useEffect(() => {
-        setUser({ name: extractName(defaultName) });
-    }, [user]);
+        setUserNameFromCookie(extractName());
+    });
```

Comment:
Trenger vi `userNameFromCookie` å settes ved hver render?

Reply:
Nei, det er jo ikke nødvendig, det trengs vel når verdien i cookien er satt/endret. Jeg fant ikke noen måte å fange opp det på. Vurderte å sammenligne navn fra cookie mot navn i state, det er kanskje bedre enn å sette navnet hver render?

Reply:
For eksempel:
```
    useEffect(() => {
        const fromCookie = extractName();
        if (fromCookie !== userNameFromCookie) {
            setUserNameFromCookie(extractName());
        }
    });
```

Reply:
Vil det fungere å kun kalle den ved mount? F.eks:
```javascript
useEffect(() => {
    setUsetNameFromCookie(extractName());
}, []);
```
Tror uansett ikke det gjør noe at den kalles ved hver render siden det er en veldig billig operasjon og de komponentene som bruker hooken (HeaderBar foreløpig) ikke re-rendres spesielt mye.

Reply:
Hvis vi bare kaller den ved mount vil vi vel ikke få oppdatert brukernavnet når det blir tilgjengelig i cookien etter at bruker har logget på?

Vi må kanskje også holde et øye med levetid på cookien. Om bruker av en eller annen årsak ikke får logget inn, men navnet fortsatt vises grunnet en gammel cookie, kan det sikkert være forvirrende.

Reply:
Mulig jeg har misforstått, men vil ikke bruker bli redirected til en login-side og så bli sendt tilbake til speil etter å ha logget inn? Da vil jo navnet være oppdatert i cookien og bli hentet ut ved mount.

Reply:
Jo, det er nok riktig at vi kan forutsette at komponenten alltid vil bli mountet på nytt etter at bruker har logget inn.

Reply:
Kall til `/` og `/static` (altså refresh i browseren) med utgått token får redirect. Kall til andre endepunkter (XHR) får 401, der må klienten selv vurdere hva som skal skje videre

Reply:
Høres ut som vi kanskje trenger `AuthContext` igjen 😅 Da kan vi håndtere 401 ved å oppdatere context slik at de komponentene som bryr seg om det kan reagere på det, som f.eks. `useLoggedInUser`